### DISCUSSION TITLE
How component render is triggered?

### DISCUSSION THREAD

**Discussion by roeycohen** [mobxjs/mobx#2675] [Q&A]
Hi!

Going briefly over mobx react code, I wasn't able to understand how an observer component is getting rerendered on observable change.

I thought that since state updating already triggers rendering, mobx is using this mechanism, but couldn't find anything related to that..

Can anyone give me a brief explanation to that matter?

10q! 🙏

**Comment by mweststrate** [comment]
https://mobx.js.org/understanding-reactivity.html

**Reply by roeycohen** [reply]
Hi @mweststrate, I'm looking for a more under the hood answer..
Let's say the observer code gets notified about a change of data, how does it triggers the render function?
🙏

**Comment by mweststrate** [comment]
It increments a counter that is stored in state / useState

**Comment by roeycohen** [comment]
Thank you @mweststrate,
I was looking for such code but wasn't able to detect it.
If not to hard, Could you please attach a link to the code itself?

P.s.
Thanks for replying! It's not obvious!

**Reply by danielkcz** [reply]
It's hidden inside useObserver internal hook within mobx-react-lite package.

**Comment by mweststrate** [comment]
search for useState. check out the sources locally if you need

**Comment by roeycohen** [comment]
Ok, looking at the code of mobx-react-lite, 
I can see that the original component is wrapped (observer.ts) with useObserver (useObserver.ts) which calls forceUpdate when a change occur.

One more thing @mweststrate, the link here is no longer valid: https://github.com/mobxjs/mobx-react-lite/commit/fc17d20e2c0826ec64bb7270aec53f8412f23b9c
is that mean that mobx 4 and 5 will no longer get update and everyone should update asap?

10x again.

**Reply by danielkcz** [reply]
That is right, MobX V4 / V5 is no longer supported and won't receive any updates unless something critical (like security issue) occurs.

But there is an automated tool called "undecorate" that makes migration very easy. See https://michel.codes/blogs/mobx6

**Reply by roeycohen** [reply]
I work with mobx 6 on another project.. it's really weird working without decorators specially if not all variables in the store are observables :)
